### PR TITLE
Add preinclude check to examples build

### PR DESCRIPTION
**Pull request by tjquinno**
We want `README.adoc` files at the top level of any project to be formatted for proper display on GitHub (which does not currently handle `include::`).

These changes add a check to the examples directory (and submodules within) to make sure that any `README.adoc` at the top level is preprocessed so included text is inserted into the file itself.

There is a new profile `adoc-check` which triggers this check. The configuration of the plug-in limits the checking to `README.adoc` files at the top level of each module.

See issues #327 and #328

### CODE REVIEW

File: examples/pom.xml [hunk 1]
```diff
@@ -69,5 +69,42 @@
                 <module>integrations</module>
             </modules>
         </profile>
+        <profile>
+            <!--
+                Pipeline builds use this profile with the generate-sources phase to convert
```
Comment (romain-grecourt) [inline examples/pom.xml]: How about we move this text under the README.md inside the /examples directory ?

**Comment by tjquinno** [comment]
Yes. They are in a later commit.

**Comment by tjquinno** [comment]
Recent build failures are due to the change to the plug-in version. Once 1.0.5 is released the pipeline for these changes should be OK again.

File: examples/pom.xml (hunk 1 above)
Comment (tjquinno) [reply examples/pom.xml]: Done.

**Pull request merged**